propolis/src/hw/pci/topology.rs

```diff
@@ -0,0 +1,427 @@
+//! A PCI topology containing one or more PCI buses.
+
+use std::collections::{BTreeMap, BTreeSet};
+use std::io::{Error as IoError, ErrorKind};
+use std::sync::{Arc, Mutex};
+
+use crate::common::RWOp;
+use crate::dispatch::DispCtx;
+use crate::inventory::{Inventory, RegistrationError};
+use crate::mmio::MmioBus;
+use crate::pio::PioBus;
+
+use super::bridge::Bridge;
+use super::{bits, Bdf, Bus, BusLocation, Endpoint, LintrCfg};
+
+use thiserror::Error;
+
+/// A logical identifier for a bus in the topology. A bus's logical identifer
```

Comment:
At the risk of being Extremely Nitpicky, I think the original's right here (AIUI most style manuals specify the `'s` pluralization except for proper nouns ending in S; AP style used to differ, but @APStylebook tweeted otherwise in 2021, and as we all know Twitter is _the_ source and summit of reasoned discourse and thoughtful communication style...).

All that said I'll gladly change it if you feel strongly :)

Reply:
Sure, feel free to keep the 's either way.